librato/__init__.py

```diff
@@ -420,6 +421,119 @@ def list_alerts(self, active_only=True, **query_props):
+
+    def get_space(self, id, **query_props):
+        """Get specific space by ID"""
+        resp = self._mexe("spaces/%s" % id,
+                          method="GET", query_props=query_props)
+        return Space.from_dict(self, resp)
+
+    def find_space(self, name):
+        if type(name) is int:
+            raise ValueError("This method expects name as a parameter, %s given" % name)
+        """Find specific space by Name"""
+        spaces = self.list_spaces(name=name)
+        # Find the Space by name (case-insensitive)
+        # This returns the first space found matching the name
+        for space in spaces:
+            if space.name and space.name.lower() == name.lower():
+                # Now use the ID to hydrate the space attributes (charts)
+                return self.get_space(space.id)
+
+        return None
+
+    def update_space(self, space, **query_props):
+        """Update an existing space (API currently only allows update of name"""
+        payload = space.get_payload()
+        for k, v in query_props.items():
+            payload[k] = v
+        resp = self._mexe("spaces/%s" % space.id,
+                          method="PUT", query_props=payload)
+        return resp
+
+    def create_space(self, name, **query_props):
+        payload = Space(self, name).get_payload()
+        for k, v in query_props.items():
+            payload[k] = v
+        resp = self._mexe("spaces", method="POST", query_props=payload)
+        return Space.from_dict(self, resp)
+
+    def delete_space(self, id):
+        """delete a space"""
+        resp = self._mexe("spaces/%s" % id, method="DELETE")
+        return resp
+
+
+    #
+    # Charts
+    #
+    def list_charts_in_space(self, space, **query_props):
+        """List all charts from space"""
+        resp = self._mexe("spaces/%s/charts" % space.id, query_props=query_props)
+        # "charts" is not in the response, but make this
+        # actually return Chart objects
+        charts = self._parse({"charts": resp}, "charts", Chart)
+        # Populate space ID
+        for chart in charts:
+            chart.space_id = space.id
+        return charts
+
+    def get_chart(self, chart_id, space_or_space_id, **query_props):
+        """Get specific chart by ID from Space"""
+        space_id = None
+        if type(space_or_space_id) is int:
+            space_id = space_or_space_id
+        elif type(space_or_space_id) is Space:
+            space_id = space_or_space_id.id
+        else:
+            raise ValueError("Space parameter is invalid")
+        # TODO: Add better handling around 404s
+        resp = self._mexe("spaces/%s/charts/%s" % (space_id, chart_id), method="GET", query_props=query_props)
+        resp['space_id'] = space_id
+        return Chart.from_dict(self, resp)
+
+    # Find a chart by name in a space. Return the first match, so if multiple
+    # charts have the same name, you'll only get the first one
```

Comment:
It may be useful here to use a parameter that modifies the default behavior and returns all the charts. With this addition the user will be able to iterate over the charts and pick the one she is interested on by inspecting the chart details (streams, etc...).